MAINTAINERS.md

```diff
@@ -0,0 +1,50 @@
+# How to release a new version
+
+## When to release a new version
+
+New version should be released when we are ready to make changes generally available.
+
+New version should not be released if we want to test our latest changes or to make them available to a limited number of users. This can be achieved without releasing a new version. Go modules allow to point to latest master or specific Git commit.
+
+Consider releasing a new Release Candidate version to make changes available to a larger group of users, if we are not ready to make them available to everyone yet.
+
+Under no circumstances, official releases must be done during the weekend or during the wee hours of the night.
+
+## Version numbers
+
+We follow the rules for semantic versioning, but prefix with the letter `v`.
+
+Examples of official releases:
+- `v1.0.0`
+- `v1.0.3`
+- `v1.2.3`
+- `v2.1.7`
+
+Examples of Release Candidates:
+- `v1.0.0-rc.4`
+- `v2.1.0-rc.1`
+
+## Step-by-step release process
+
+1. Clone or fetch latest master of these Git repositories:
+  - https://github.com/storj/storj
+  - https://github.com/storj/gateway
+  - https://github.com/storj/uplink-c
+2. For each of them update, `go.mod` and `testsuite/go.mod` to latest master (or the specific Git commit that will be tagged as a new version) of `storj.io/uplink`.
+3. Use `go mod tidy` to update the respective `go.sum` files.
+4. Push a change to Gerrit with the updated `go.mod` and `go.sum` files for each of the Git repositories.
+5. Wait for the build to finish. If the build fails for any of the Git repositories, abort the release process. Investigate the issue, fix it, and start over the release process.
+6. If all builds are successful, do not merge the changes yet.
+7. If you haven't done this yet, announce your intention to make a new release to the #libuplink Slack channel.
+8. Wait for a confirmation by at least one of the main maintainers of the project.
```

Comment:
Confirmation of what?

Reply:
That you can proceed with the next step.

Reply:
Why are the downstream users the gatekeepers of the release? What would they be confirming?

Reply:
I meant the maintainers of the _this_ (storj.io/uplink) project.

Reply:
I don't know who the "main maintainers" are. I assumed I was one of them, but maybe I'm not. Should we have a list?

Reply:
Well, you are, but you still have to wait for a confirmation by another maintainer. As anyone else too.

The idea is similar to getting approvals in code reviews. Releasing a new version should not be done lightly and it would be better to have more eyes before pushing the button to ensure we are not missing anything important.

I am not sure if we should do a list of people. We have to maintain it if we do.